src/app/util/ember-compatibility-functions.cpp

```diff
@@ -94,5 +170,143 @@ bool ServerClusterCommandExists(chip::ClusterId aClusterId, chip::CommandId aCom
     return emberAfContainsServer(aEndPointId, aClusterId);
 }
 
+CHIP_ERROR ReadSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVWriter * apWriter, bool * apDataExists)
+{
+    // TODO: The buffer can hold values as large as EMBER_AF_MAXIMUM_SEND_PAYLOAD_LENGTH, the ember allows an attribute as large as
+    // 2^16 bytes while CHIP Spec says the single reported data should fits into MTU.
+    static uint8_t data[EMBER_AF_MAXIMUM_SEND_PAYLOAD_LENGTH];
+
+    ChipLogDetail(DataManagement,
+                  "Received Cluster Command: Cluster=%" PRIx16 " NodeId=0x" ChipLogFormatX64 " Endpoint=%" PRIx8 " FieldId=%" PRIx16
+                  " ListIndex=%" PRIx16,
+                  aClusterInfo.mClusterId, ChipLogValueX64(aClusterInfo.mNodeId), aClusterInfo.mEndpointId, aClusterInfo.mFieldId,
+                  aClusterInfo.mListIndex);
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    EmberAfAttributeType dataType;
+
+    EmberAfAttributeMetadata * metadata = NULL;
+    EmberAfAttributeSearchRecord record;
+    EmberAfStatus status;
+    record.endpoint         = aClusterInfo.mEndpointId;
+    record.clusterId        = aClusterInfo.mClusterId;
+    record.clusterMask      = CLUSTER_MASK_SERVER;
+    record.attributeId      = aClusterInfo.mFieldId;
+    record.manufacturerCode = 0;
+    status                  = emAfReadOrWriteAttribute(&record, &metadata, data, sizeof(data), false); // write?
+
+    if (apDataExists != nullptr)
+    {
+        *apDataExists = (EMBER_ZCL_STATUS_SUCCESS == status);
+    }
+
+    VerifyOrExit(apWriter != nullptr, /* no op */);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        // TODO: replace by enum after #7482
+        err = apWriter->Put(chip::TLV::ContextTag(AttributeDataElement::kCsTag_Status),
+                            Protocols::InteractionModel::ToUint16(Protocols::InteractionModel::ProtocolCode::UnsupportedAttribute));
```

Comment:
Similarly to the previous comment, there can be a few different statuses. It does not necessarily mean `UnsupportedAttribute`